openssl/src/x509/store.rs

```diff
@@ -86,6 +90,40 @@ impl X509StoreBuilderRef {
         unsafe { cvt(ffi::X509_STORE_add_cert(self.as_ptr(), cert.as_ptr())).map(|_| ()) }
     }
 
+    /// Sets the maximum verification depth, or the maximum number of intermediate CA certificates that can appear in a chain.
+    pub fn set_depth(&mut self, depth: i32) -> Result<(), ErrorStack> {
+        unsafe { cvt(ffi::X509_STORE_set_depth(self.as_ptr(), depth)).map(|_| ()) }
+    }
+
+    /// Sets the verification flags used to verify a certificate against a chain.
+    pub fn set_flags(&mut self, flags: X509VerifyFlags) -> Result<(), ErrorStack> {
+        unsafe { cvt(ffi::X509_STORE_set_flags(self.as_ptr(), flags.bits())).map(|_| ()) }
+    }
+
+    /// Sets the purpose used to verify the certificate chain.
+    pub fn set_purpose(&mut self, purpose: PurposeId) -> Result<(), ErrorStack> {
+        unsafe { cvt(ffi::X509_STORE_set_purpose(self.as_ptr(), purpose.as_raw())).map(|_| ()) }
+    }
+
+    /// Sets the trust value used to verify the certificate chain.
+    pub fn set_trust(&mut self, trust: TrustId) -> Result<(), ErrorStack> {
+        unsafe { cvt(ffi::X509_STORE_set_trust(self.as_ptr(), trust.as_raw())).map(|_| ()) }
+    }
+
+    /// Load trusted certificate(s) into the `X509Store` from a file or a directory.
+    ///
+    /// The certificates in the file or directory should be in a hashed format.
+    pub fn load_locations(&mut self, file: &str, dir: &str) -> Result<(), ErrorStack> {
+        unsafe {
+            cvt(ffi::X509_STORE_load_locations(
+                self.as_ptr(),
+                file.as_ptr() as *const c_char,
```

Comment:
Rust strings are not null terminated. This should also take Path arguments like https://github.com/sfackler/rust-openssl/blob/134a9de0f071cbbe4889a0ad7b040c234a76b7ae/openssl/src/x509/mod.rs#L873.